osu.Game/Screens/Play/Player.cs

```diff
@@ -1085,7 +1085,10 @@ public override bool OnExiting(ScreenExitEvent e)
 
                 // if arriving here and the results screen preparation task hasn't run, it's safe to say the user has not completed the beatmap.
                 if (prepareScoreForDisplayTask == null)
+                {
+                    BreakOverlay.Info.GradeDisplay.Current.UnbindFrom(ScoreProcessor.Rank);
```

Comment:
This can't be in `Player.cs`. At most, this could be a method call on `BreakOverlay`, but I'm not sure this is the correct place to fix this. Would it not make more sense that `ScoreProcessor`'s rank doesn't change after the exit process beings, rather than just hiding the fact that it changed from being displayed?

Reply:
I could move the logic in `Player` for failing on early exit over to `SubmittingPlayer` so that it only effects the submission itself.